editor/modes/visual-editor/block.js

```diff
@@ -135,19 +175,23 @@ class VisualEditorBlock extends wp.element.Component {
 }
 
 export default connect(
-	( state, ownProps ) => ( {
-		order: state.blocks.order.indexOf( ownProps.uid ),
-		block: state.blocks.byUid[ ownProps.uid ],
-		isSelected: state.selectedBlock.uid === ownProps.uid,
-		isHovered: state.hoveredBlock === ownProps.uid,
-		focus: state.selectedBlock.uid === ownProps.uid ? state.selectedBlock.focus : null,
-		isTyping: state.selectedBlock.uid === ownProps.uid ? state.selectedBlock.typing : false,
-	} ),
+	( state, ownProps ) => {
+		const order = state.blocks.order.indexOf( ownProps.uid );
+		return {
+			previousBlock: order === 0 ? null : state.blocks.byUid[ state.blocks.order[ order - 1 ] ],
```

Comment:
Maybe this should be `<= 0` to account for the chance that `uid` is not present in the `order` array?

Alternatively perhaps the fallback could occur as an `||` case:

```js
previousBlock: state.blocks.byUid[ state.blocks.order[ state.blocks.order.indexOf( ownProps.uid ) - 1 ] ] || null
```

(with better formatting)